pkg/archive/copy.go

```diff
@@ -218,11 +252,11 @@ func CopyInfoDestinationPath(path string) (info CopyInfo, err error) {
 		// Ensure destination parent dir exists.
 		dstParent, _ := SplitPathDirEntry(path)
 
-		parentDirStat, err := os.Lstat(dstParent)
```

Comment:
I might be forgetting something, but wouldn't simply changing this from `os.Lstat` to `os.Stat` fix this issue? `os.Stat(name)` will follow symlinks and you'll get the `os.Fileinfo` describing the target.

Not only would this be a much shorter change but it also covers the case where `dstParent` is a symlink to another symlink (to possibly yet another symlink) that finally points to a directory. That is a test case that is not currently covered but looks like would still fail under this current fix.